IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m

```diff
@@ -0,0 +1,131 @@
+#import "ASAuthorizationSingleSignOnProvider+MSIDExtensions.h"
+#import "MSIDSSOExtensionInteractiveTokenRequest.h"
+#import "MSIDInteractiveTokenRequest+Internal.h"
+#import "MSIDJsonSerializer.h"
+#import "MSIDInteractiveRequestParameters.h"
+#import "MSIDAccountIdentifier.h"
+#import "MSIDAuthority.h"
+#import "MSIDSSOExtensionTokenRequestDelegate.h"
+#import "MSIDBrokerOperationInteractiveTokenRequest+InteractiveParameters.h"
+#import "MSIDBrokerOperationInteractiveTokenRequest+SSORequest.h"
+
+@interface MSIDSSOExtensionInteractiveTokenRequest () <ASAuthorizationControllerPresentationContextProviding>
+
+@property (nonatomic) ASAuthorizationController *authorizationController;
+@property (nonatomic, copy) MSIDInteractiveRequestCompletionBlock requestCompletionBlock;
+@property (nonatomic) MSIDSSOExtensionTokenRequestDelegate *extensionDelegate;
+@property (nonatomic) ASAuthorizationSingleSignOnProvider *ssoProvider;
+
+@end
+
+@implementation MSIDSSOExtensionInteractiveTokenRequest
+
+- (instancetype)initWithRequestParameters:(MSIDInteractiveRequestParameters *)parameters
+                             oauthFactory:(MSIDOauth2Factory *)oauthFactory
+                   tokenResponseValidator:(MSIDTokenResponseValidator *)tokenResponseValidator
+                               tokenCache:(id<MSIDCacheAccessor>)tokenCache
+                     accountMetadataCache:(MSIDAccountMetadataCacheAccessor *)accountMetadataCache
+{
+    self = [super initWithRequestParameters:parameters
+                               oauthFactory:oauthFactory
+                     tokenResponseValidator:tokenResponseValidator
+                                 tokenCache:tokenCache
+                       accountMetadataCache:accountMetadataCache];
+
+    if (self)
+    {
+        _extensionDelegate = [MSIDSSOExtensionTokenRequestDelegate new];
+        _extensionDelegate.context = parameters;
+        __weak typeof(self) weakSelf = self;
+        _extensionDelegate.completionBlock = ^(MSIDTokenResponse *response, NSError *error)
+        {
+            [weakSelf handleTokenResponse:response error:error completionBlock:weakSelf.requestCompletionBlock];
+        };
+        
+        _ssoProvider = [ASAuthorizationSingleSignOnProvider msidSharedProvider];
+    }
+
+    return self;
+}
+
+#pragma mark - MSIDInteractiveTokenRequest
+
+- (void)executeRequestWithCompletion:(MSIDInteractiveRequestCompletionBlock)completionBlock
+{
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Beginning interactive broker flow.");
+    
+    if (!completionBlock)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, self.requestParameters, @"Passed nil completionBlock. End silent broker flow.");
+        return;
+    }
+    
+    NSString *upn = self.requestParameters.accountIdentifier.displayableId ?: self.requestParameters.loginHint;
+    
+    [self.requestParameters.authority resolveAndValidate:self.requestParameters.validateAuthority
```

Comment:
shouldn't this be common in both interactive request cases?

Reply:
It is hard to share this logic because nested block is a bit different. Agreed to keep as it is for now.